ros_buildfarm/workspace.py

```diff
@@ -65,4 +107,4 @@ def call_catkin_make_isolated(
 
     print("Invoking '%s' in '%s'" % (cmd, workspace_root))
     return subprocess.call(
```

Comment:
A couple of things I noticed when messing with this for CI purposes in another branch. I'm not sure if they should be changed here, or if I'm doing something wrong:
1. I needed to set `COLCON_CURRENT_PREFIX` before each setup file to make this work.
2. In the hook files `typeset` is called, which is a bash alias. So this `subprocess.call()` had to get `executable='/bin/bash'` to make that work right.

Reply:
Clarification for (2):

After working around (1), whenever I invoke a colcon operation, I see messages like this for many (every?) package in the parent result space:
```
/bin/sh: 23: /home/cottsay/ws/ros2-linux-official/share/<package_name_here>/hook/cmake_prefix_path.sh: typeset: not found
```

Since `typeset` is a built-in bash command [[1](https://linux.die.net/man/1/typeset)], I concluded that `sh` was being used instead of bash. These errors don't seem to be fatal for the operations I'm performing, but clearly something isn't working right.

I was able to work around this issue by changing the shell to `/bin/bash` instead of the default `sh`:
```diff
--- a/ros_buildfarm/workspace.py
+++ b/ros_buildfarm/workspace.py
@@ -102,4 +102,4 @@ def call_build_tool(
 
     print("Invoking '%s' in '%s'" % (cmd, workspace_root))
     return subprocess.call(
-        cmd, cwd=workspace_root, shell=True, stderr=subprocess.STDOUT, env=env)
+        cmd, cwd=workspace_root, shell=True, stderr=subprocess.STDOUT, env=env, executable='/bin/bash')
```

[1] https://linux.die.net/man/1/typeset

Reply:
1. Yes, it is a limitation of `sh` to not be able to detect its own locations and therefore if a prefix has been relocated. This will certainly need an update (either to `bash` or setting the prefix explicitly as you suggested). Since it isn't needed for this PR I would keep for another PR - maybe within the CI job one.

2. Could this line be the source of the problem: https://github.com/colcon/colcon-core/blob/ce54acd21ef3d337a6bad777887cf63fda6a244c/colcon_core/shell/template/package.sh.em#L23 If yes, then this should be addressed in the shell template in `colcon-core`.

Reply:
I think your analysis is accurate - thanks for following up! :+1: